lib/honeycomb/propagation/honeycomb.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require "base64"
+require "json"
+require "uri"
+
+module Honeycomb
+  # Parsing and propagation for honeycomb trace headers
+  module HoneycombPropagation
+    # Parse trace headers
+    module UnmarshalTraceContext
+      def parse(serialized_trace)
+        unless serialized_trace.nil?
+          version, payload = serialized_trace.split(";", 2)
+
+          if version == "1"
+            trace_id, parent_span_id, trace_fields, dataset = parse_v1(payload)
+
+            if !trace_id.nil? && !parent_span_id.nil?
+              return [trace_id, parent_span_id, trace_fields, dataset]
+            end
+          end
+        end
+
+        [nil, nil, nil, nil]
+      end
+
+      def parse_v1(payload)
+        trace_id, parent_span_id, trace_fields, dataset = nil
+        payload.split(",").each do |entry|
+          key, value = entry.split("=", 2)
+          case key
```

Comment:
Should we `key.downcase` here too to ignore casing?

Reply:
oooh I like that